lib/galaxy/datatypes/display_applications/parameters.py

```diff
@@ -107,8 +109,12 @@ def prepare(self, other_values, dataset_hash, user_hash, trans):
             # start conversion
             # FIXME: Much of this is copied (more than once...); should be some abstract method elsewhere called from here
             # find target ext
-            target_ext, converted_dataset = data.find_conversion_destination(self.formats, converter_safe=True)
-            if target_ext and not converted_dataset:
+            direct_match, target_ext, converted_dataset = data.find_conversion_destination(self.formats, converter_safe=True)
+            # TODO: Q: I guess we could skip the first if branch but code might
+            # be more readable
+            if direct_match:
```

Comment:
I'd say `if not direct_match:` and then have the other 2 if statements below seems more readable ?

Reply:
OK.